src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -66,68 +66,181 @@ public static int IndexOf(ref char searchSpace, int searchSpaceLength, ref char
         }
 
         [MethodImpl(MethodImplOptions.AggressiveOptimization)]
-        public static unsafe int SequenceCompareTo(ref char first, int firstLength, ref char second, int secondLength)
+        public static unsafe int SequenceCompareTo(ref char firstStart, int firstLength, ref char secondStart, int secondLength)
         {
             Debug.Assert(firstLength >= 0);
             Debug.Assert(secondLength >= 0);
 
-            int lengthDelta = firstLength - secondLength;
-
-            if (Unsafe.AreSame(ref first, ref second))
+            if (Unsafe.AreSame(ref firstStart, ref secondStart))
                 goto Equal;
 
-            IntPtr minLength = (IntPtr)((firstLength < secondLength) ? firstLength : secondLength);
-            IntPtr i = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
+            int minLength = (firstLength < secondLength) ? firstLength : secondLength;
```

Comment:
Is the asm for this as good as if it were something like:
```C#
int minLength = secondLength;
if (firstLength < secondLength)
{
    minLength = firstLength;
}
```
?

Reply:
I would go with `Math.Min` (and potentially optimize that method more).

Reply:
`Math.Min` generates same asm. The assign then test, drops a jump so slightly better.

However, will go with `Math.Min` as it would be easier to recognise for `cmov` in the hope that one day the Jit generates it.